### DISCUSSION TITLE
Expose userId, contactId and conversationId from the widget

### DISCUSSION THREAD

**Discussion by virus2016** [chatwoot/chatwoot#8652] [Feature Requests]
### Is your feature or enhancement related to a problem? Please describe.

The widget must be interoperable with the backend SDK. For instance, when a customer initiates a conversation, the website or app should be capable of appending additional details and potentially even dispatching a response to the customer upon the completion of an action.

### Describe the solution you'd like

setUser > getUser > call our api with user Id from widget > send a message to the customer from the api using the Chatwood sdk

### Describe alternatives you've considered

Use my own api to create a user first, then tell chatwood the id

**Comment by linear** [comment]
<p><a href="https://linear.app/chatwoot/issue/PR-807/expose-userid-contactid-and-conversationid-from-the-widget">PR-807 Expose userId, contactId and conversationId from the widget</a></p>
